apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -506,6 +507,8 @@ namespace MWRender
         mRootNode->getOrCreateStateSet()->setAttribute(Shader::RemovedAlphaFunc::getInstance(GL_ALWAYS));
         // The transparent renderbin sets alpha testing on because that was faster on old GPUs. It's now slower and breaks things.
         mRootNode->getOrCreateStateSet()->setMode(GL_ALPHA_TEST, osg::StateAttribute::OFF);
+        osgUtil::RenderBin::getRenderBinPrototype("DepthSortedBin")->setStateSet(nullptr);
+        osgUtil::RenderBin::getRenderBinPrototype("SORT_BACK_TO_FRONT")->setStateSet(nullptr);
```

Comment:
As mentioned in the other attempt at this change, I think accessing the renderbins by name is more fragile than explicitly disabling the alpha test. For example, the prototype could get replaced after this has run.

Reply:
In this case, we can avoid the unwanted state during creation of the new prototype.

Reply:
For a change only affecting performance on a subset of systems, is "fragile" the correct term? Surely an author of a new RenderBin has a million ways to screw up the performance unintentionally.

Reply:
First things first, disabling the forced alpha test isn't a performance optimisation. There were some graphical artefacts, hence the comment for the old approach saying it was *now slower and breaks things*. Based on the comment I added to the MR on GitLab right after pushing it, it broke rain (which doesn't use shaders ever yet).

The author of a new renderbin isn't necessarily us - it could be added on OSG's end, and also they might change the values of the strings - because Robert refuses to add OpenMW to his regression tests, there have been several cases of him breaking things by assuming no OSG apps care about a change, despite us relying on it (although he's also rejected loads of bug fixes from us in case they break theoretical OSG apps that relied on the old behaviour, which is annoying). I don't think it's likely, but it's just generally good practice to at least make sure any such errors are build-time if possible.

I take it these strings aren't also available via an enum or anything like that?

One thing that might work would be to keep the old approach of explicitly disabling the alpha test, too (I'm aware I just made you remove it). We've already got to have a stateset there, so the impact should be minimal, but it would give us something to fall back to if anything else changed.

I'd roughly expect this PR to decrease the stategraph count by one for each non-depth RTT pass, which doesn't seem like a big gain if it ends up causing us hassle in the future, so I'd rather get the potential hassle to a minimum.

Reply:
There is no enum to use. Well, I find it hard to believe the strings would change. Such change would break every application ever made with osg because we have to refer by name to use the RenderBins.

Reply:
> First things first, disabling the forced alpha test isn't a performance optimisation. There were some graphical artefacts, hence the comment for the old approach saying it was now slower and breaks things.

In this case, I prefer to keep the GL_ALPHA_TEST mode removed as is currently the case in this PR. We will notice any breakage related to the renderbin if it happens.

Reply:
> There is no enum to use. Well, I find it hard to believe the strings would change. Such change would break every application ever made with osg because we have to refer by name to use the RenderBins.

There's also the hint thing on the stateset that takes an enum, so it wouldn't be *every* OSG app, although that's older and not recommended anymore. Between initially typing this out and realising I'd never sent it, I did actually find the enum I was thinking of (`Renderbin::SortMode`) while I was looking into the stategraph counting, but it looks like you can't actually do what we want with it. Anyway, strings are as good as this MR's going to get.